pug/index.d.ts

```diff
@@ -16,35 +16,124 @@ declare module 'pug' {
-    export function renderFile(path: string, options?: Options, callback?: (err: Error, html: string) => void): string;
 
-    // else
-    export type ClientFunctionString = string;  // ex: 'function (locals) {...}'
-    export type compileTemplate = (locals?: any) => string;
+    /**
+     * Compile a Pug template file to a string of JavaScript that can be used client side along with the Pug runtime.
+     */
+    export function compileFileClient(path: string, options?: Options): string;
+
+    /**
+     * Compile a Pug template and render it without locals to html string.
+     */
+    export function render(template: string): string;
+    /**
+     * Compile a Pug template and render it with locals to html string.
+     * @param {(Options & LocalsObject)} options Pug Options and rendering locals
+     */
+    export function render(template: string, options: Options & LocalsObject): string;
+    /**
+     * Compile a Pug template and render it without locals to html string.
+     */
+    export function render(template: string, callback: (err: Error | null, res: string) => void): void;
+    /**
+     * Compile a Pug template and render it with locals to html string.
+     * @param {(Options & LocalsObject)} options Pug Options and rendering locals
+     * @param {((err: Error | void, res: string) => void)} callback Node.js-style callback receiving the rendered results. This callback is called synchronously.
+     */
+    export function render(template: string, options: Options & LocalsObject, callback: (err: Error | null, res: string) => void): void;
+
+    /**
+     * Compile a Pug template from a file and render it without locals to html string.
+     */
+    export function renderFile(path: string): string;
+    /**
+     * Compile a Pug template from a file and render it with locals to html string.
+     * @param {(Options & LocalsObject)} options Pug Options and rendering locals
+     */
+    export function renderFile(path: string, options: Options & LocalsObject): string;
+    /**
+     * Compile a Pug template from a file and render it without locals to html string.
+     */
+    export function renderFile(path: string, callback: (err: Error | null, res: string) => void): void;
+    /**
+     * Compile a Pug template from a file and render it with locals to html string.
+     * @param {(Options & LocalsObject)} options Pug Options and rendering locals
+     * @param {((err: Error | void, res: string) => void)} callback Node.js-style callback receiving the rendered results. This callback is called synchronously.
+     */
+    export function renderFile(path: string, options: Options & LocalsObject, callback: (err: Error | null, res: string) => void): void;
+
+    /////////////////////////
+    ///    Interfaces     ///
+    /////////////////////////
+
+    /** 
+     * A function that can be use to render html string of compiled template.
+     */
+    export interface compileTemplate {
```

Comment:
`export type compileTemplate = (locals?: any) => string;` was simpler. As of Microsoft/TypeScript#11540 you should be able to get jsdoc info from type aliases.